### ISSUE TITLE
different behaviour in modelio when use aggregation vs composition

### ISSUE THREAD

**Issue by dpalomar** [jhipster/jhipster-uml#234]
Really I don't know if this is and bug or it's my fault.

I'm using modelio 3.5.

Following the examples of documentation, the docs say: _our parser system doesn’t care if you use an aggregation or a composition. You can use whatever you want._ But I've found some differences.

If I use composition relationship between Employee and Job with OneToMany (unidirectional) the parser does not show the warning message that shows when I use the same relationship with aggregation, and write a `@OneToOne` relationship in the code, but using aggregation the code is `@OneToMany`.

Another case is using one-to-one aggregation relationship, the generated code is `@ManyToOne`.

So it appear that the parser always generates one-to-one when composition is used, and many-to-... or ...-to-many when aggregation is used.

This occurs with modelio, with umldesigner the result is not the same, but different.

**Comment by MathieuAA** [comment]
I guess I should update the docs then... For a one-to-one, composition is to be used. Otherwise, an aggregation may always be needed... Thanks for taking the time to report this

Le 11 novembre 2016 14:25:39 GMT+01:00, David Palomar <notifications@github.com> a écrit :
>Really I don't know if this is and bug or it's my fault.
>
>I'm using modelio 3.5.
>
>Following the examples of documentation, the docs say: _our parser
>system doesn’t care if you use an aggregation or a composition. You can
>use whatever you want._ But I've found some differences.
>
>If I use composition relationship between Employee and Job with
>OneToMany (unidirectional) the parser does not show the warning message
>that shows when I use the same relationship with aggregation, and write
>a `@OneToOne` relationship in the code, but using aggregation the code
>is `@OneToMany`.
>
>Another case is using one-to-one aggregation relationship, the
>generated code is `@ManyToOne`.
>
>So it appear that the parser always generates one-to-one when
>composition is used, and many-to-... or ...-to-many when aggregation is
>used.
>
>This occurs with modelio, with umldesigner the result is not the same,
>but different.
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/jhipster/jhipster-uml/issues/234

-- 
Mathieu ABOU-AICHI

**Comment by MathieuAA** [comment]
By the way, it is said that the parser doesn't care if the parser is the UMLDesigner one... not for Modelio. But I'm updating it anyway for best practice.

**Comment by dpalomar** [comment]
TL;DR

Yep it's true, it's under umldesigner section, but the complete paragraph mentions Modelio also (_As mentioned with Modelio, our parser system doesn’t care ..._) and in the Modelio section there is a similar phrase _"You can use either of them, the parser will just see the two as associations anyway."_

The following paragraph under modelio section explains in detail the purpose of both relationships, but I think that it's bypassed (my fault):

"_Notice the left panel entitled ‘Class Model’. You only need the ‘Class’, ‘Attributes’, ‘Aggregation’, ‘Composition’ and ‘Data Type’ objects. You can already guess why you need the first 2. The ‘Aggregation’ object is used to defined aggregations: B is aggregated to A means that A has 0, 1 or more (n) instances of B. A doesn’t create (and destroy) B instances. Composition means that if A is composed of B (0, 1, or n instances), then it creates, manages and destroys B instances._"

Perhaps it should be summarized like: "if you wants a one-to-one relationship use composition and aggregation in other case".  :-P 

I'm sorry about that, but I've spent several days to find where the problem was.

**Comment by MathieuAA** [comment]
Don't be. Thanks again for notifying me

Le 12 novembre 2016 00:12:28 GMT+01:00, David Palomar <notifications@github.com> a écrit :
>TL;DR
>
>Yep it's true, it's under umldesigner section, but the complete
>paragraph mentions Modelio also (_As mentioned with Modelio, our parser
>system doesn’t care ..._) and in the Modelio section there is a similar
>phrase _"You can use either of them, the parser will just see the two
>as associations anyway."_
>
>The following paragraph under modelio section explains in detail the
>purpose of both relationships, but I think that it's bypassed (my
>fault):
>
>"_Notice the left panel entitled ‘Class Model’. You only need the
>‘Class’, ‘Attributes’, ‘Aggregation’, ‘Composition’ and ‘Data Type’
>objects. You can already guess why you need the first 2. The
>‘Aggregation’ object is used to defined aggregations: B is aggregated
>to A means that A has 0, 1 or more (n) instances of B. A doesn’t create
>(and destroy) B instances. Composition means that if A is composed of B
>(0, 1, or n instances), then it creates, manages and destroys B
>instances._"
>
>Perhaps it should be summarized like: "if you wants a one-to-one
>relationship use composition and aggregation in other case".  :-P 
>
>I'm sorry about that, but I've spent several days to find where the
>problem was.
>
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/jhipster/jhipster-uml/issues/234#issuecomment-260077148

-- 
Mathieu ABOU-AICHI
